rdmd.d

```diff
@@ -916,3 +898,78 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+struct FileAttributes
+{
+    uint attributes;
+    bool exists() const { return attributes != 0xFFFFFFFF; }
+    bool isFile() const
+        in { assert(exists); } do
+    {
+        import std.file : attrIsFile;
+        return attrIsFile(attributes);
+    }
+}
+
+struct Yap
+{
```

Comment:
Move `std.file` imports here. That way, it will not be possible to accidentally use `std.file` functions bypassing logging.

Reply:
There's still a fair amount of `std.file` functions used in the rest of the file
```
thisExePath
tempDir
remove
FileException
DirEntry
dirEntries
```

Reply:
Please use selective imports for that.

`remove` is sticking out: should it be called via the wrapper instead? If not, then it should be a static or fully-qualified import to make it explicit.

Reply:
The `remove` case is awkward.  It has a try/catch block with logic to handle what happens if the file is being used.  It's in the first part of the `rebuild` function.

I can switch them to selective imports if you think that's better, but none of the other modules use selective imports, so I wasn't sure if I should follow convention or do the "safer" thing.

Reply:
Yes, please use selective imports. Using selective (or static) imports provides half of the justification for the filesystem operations wrapper: to prevent accidentally bypassing it. Without that, it's more difficult to justify this entire pull request.

Reply:
The `remove` call is already accompanied by a `yap`, so with a bit of refactoring of the `!dryRun` part it looks like a good candidate of being converted to a call to the wrapper.

Reply:
`dirEntries` should also probably be moved to the wrapper, since it's not much different from `stat` in principle.

Reply:
ok, done except moving the remove case to a wrapper.  have another look at the current set of changes in #299 and let me know what you think